### PR TITLE
Enabling Windows operational readiness tests on CAPZ

### PR DESCRIPTION
**Pull request by knabben**
In order to use the tests in the repo, this PR intends to enable the custom Windows test validation on a CAPZ cluster.

### CODE REVIEW

**Review by dougsland — COMMENTED** [review]
LGTM

**Comment by marosset** [comment]
@knabben You need to add in a refernce to the `cluster-api-provider-azure` repo like https://github.com/kubernetes/test-infra/blob/744e04bab59116dca3ad19c8116e7fd11c0573bd/config/jobs/kubernetes-sigs/sig-windows/soak-tests.yaml#L23-L27
(note: workdir=true is required too)
This ensures everything up to and including the ci-entrypoint.sh script is available

File: config/jobs/kubernetes-sigs/sig-windows/windows-op-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            - name: DEPLOY_AZURE_CSI_DRIVER
+              value: "true"
+            - name: AZUREDISK_CSI_DRIVER_VERSION
+              value: "master"
+            - name: PROMETHEUS_STORAGE_CLASS_PROVISIONER
+              value: "kubernetes.io/azure-disk"
+            - name: PROMETHEUS_STORAGE_CLASS_VOLUME_TYPE
+              value: "StandardSSD_LRS"
+    annotations:
+      testgrid-dashboards: sig-windows-signal
```
Comment (marosset) [inline config/jobs/kubernetes-sigs/sig-windows/windows-op-tests.yaml:66]: Add this to `sig-windows-presubmit` and not `sig-windows-signal`

File: config/jobs/kubernetes-sigs/sig-windows/windows-op-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          securityContext:
+            privileged: true
+          env:
+            # CAPZ variables
+            - name: NODE_MACHINE_TYPE
+              value: "Standard_D4s_v3"
+            - name: TEST_WINDOWS
+              value: "true"
+            - name: KUBERNETES_VERSION
+              value: "v1.23.5"
```
Comment (marosset) [inline config/jobs/kubernetes-sigs/sig-windows/windows-op-tests.yaml:38]: This should match the branch for the kubekins image we use above
If we want to test on master we can leave the container image on master and set `KUBERNETES_VERSION=latest`

**Pull request merged**